### PR TITLE
implementation of egress rules with one domain

### LINKED ISSUE #1180: Re-enable routing rules for egress domains
Support for external services through egress rules has omitted routing rules/destination policies on these rules. Egress rule should be considered as just another service spec and so, the destination in route/destinationpolicy should be matched with the destination in the egress rule to generate the appropriate settings for timeouts/retries/circuit breakers, etc.

Without this, apps will not be able to use istio routing rules or any of Envoy's features when talking to external services.

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
**What this PR does / why we need it**:
Implements handling egress rules according to the new format - one domain, using IstioService for destination

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1180

**Special notes for your reviewer**:

**Release note**:

```
```

### CODE REVIEW

File: proxy/envoy/config.go [hunk 1]
```diff
@@ -658,23 +658,27 @@ func appendPortToDomains(domains []string, port int) []string {
...
+	// the name of the virtual host and of the original dst cluster
+	name := domain + ":" + strconv.Itoa(port.Port)
 
 	if rule.UseEgressProxy {
 		externalTrafficCluster = buildOutboundCluster("istio-egress", port, nil)
 		externalTrafficCluster.ServiceName = ""
 		externalTrafficCluster.Type = ClusterTypeStrictDNS
 		externalTrafficCluster.Hosts = []Host{{URL: fmt.Sprintf("tcp://%s", mesh.EgressProxyAddress)}}
 	} else {
-		externalTrafficCluster = buildOriginalDSTCluster("orig-dst-cluster"+protocolSuffix, mesh.ConnectTimeout)
```
Comment (rshriram) [inline proxy/envoy/config.go]: Please keep this if possible. We need to deduplicate the number of orig_dst_clusters

**Comment by GregHanson** [comment]
```=================== Passed all tests! =====================
tests: [http-reachability http2-reachability tcp-reachability ingress egress routing-rules zipkin], count: 1

I0907 13:51:12.647319    6630 driver.go:155] 

=================== Passed infrastructure tests! =====================
([]main.infra) (len=2 cap=2) {
 (main.infra) {
  Name: (string) (len=15) "(default infra)",
  Hub: (string) (len=8) "gihanson",
  Tag: (string) (len=26) "dev_170907_Sep091504809883",
  MixerImage: (string) (len=67) "gcr.io/istio-testing/mixer:986c6e1eb9c7dcff42e3df5aefa9ce858ece885e",
  CaImage: (string) (len=70) "gcr.io/istio-testing/istio-ca:2baec6baacecbd516ea0880573b6fc3cd5736739",
  Namespace: (string) "",
  IstioNamespace: (string) "",
  Verbosity: (int) 2,
  apps: (map[string][]string) <nil>,
  Auth: (istio_proxy_v1_config.ProxyMeshConfig_AuthPolicy) NONE,
  Mixer: (bool) true,
  Ingress: (bool) true,
  Egress: (bool) true,
  Zipkin: (bool) true,
  DebugPort: (int) 0,
  checkLogs: (bool) true,
  namespaceCreated: (bool) false,
  istioNamespaceCreated: (bool) false,
  UseInitializer: (bool) false,
  InjectConfig: (*inject.Config)(<nil>),
  config: (model.IstioConfigStore) <nil>
 },
 (main.infra) {
  Name: (string) (len=12) "(auth infra)",
  Hub: (string) (len=8) "gihanson",
  Tag: (string) (len=26) "dev_170907_Sep091504809883",
  MixerImage: (string) (len=67) "gcr.io/istio-testing/mixer:986c6e1eb9c7dcff42e3df5aefa9ce858ece885e",
  CaImage: (string) (len=70) "gcr.io/istio-testing/istio-ca:2baec6baacecbd516ea0880573b6fc3cd5736739",
  Namespace: (string) "",
  IstioNamespace: (string) "",
  Verbosity: (int) 2,
  apps: (map[string][]string) <nil>,
  Auth: (istio_proxy_v1_config.ProxyMeshConfig_AuthPolicy) MUTUAL_TLS,
  Mixer: (bool) true,
  Ingress: (bool) true,
  Egress: (bool) true,
  Zipkin: (bool) true,
  DebugPort: (int) 0,
  checkLogs: (bool) true,
  namespaceCreated: (bool) false,
  istioNamespaceCreated: (bool) false,
  UseInitializer: (bool) false,
  InjectConfig: (*inject.Config)(<nil>),
  config: (model.IstioConfigStore) <nil>
 }
}
```

**Comment by rshriram** [comment]
tested on two private clusters. merging..

**Pull request merged**